R/quo.R

```diff
@@ -411,10 +412,23 @@ as_quosure <- function(x, env = caller_env()) {
     new_quosure(x, env)
   }
 }
+
 #' @export
 print.quosure <- function(x, ...) {
-  x <- set_attrs(x, class = "formula")
-  NextMethod()
+  env_type <- env_type(get_env(x))
+
+  if (env_type %in% c("frame", "local")) {
+    addr <- sxp_address(get_env(x))
```

Comment:
I have mixed feelings about printing the environment address — it doesn't add much information and adds spurious diffs when re-rendering markdown.  OTOH it does help indicate that evaluating too things that otherwise look identical will yield different results.